### PR TITLE
take care of Delegators when working with Sets/Maps

### PR DESCRIPTION
**Pull request by rbri**
Sorry for having a Delegator problem again (see https://github.com/HtmlUnit/htmlunit/issues/373 for details)

Have written some tests and a first implementation - waiting for feedback

### CODE REVIEW

**Comment by gbrail** [comment]
This looks reasonable to me. Does anyone else who works regularly with delegators have an idea?

**Comment by SebastianGlass** [comment]
I like the way it is fixed, but just to understand... Why are there is many copy pasted static methods instead of an util class (for the new keys method for example)

**Comment by rbri** [comment]
> 
> 
> I like the way it is fixed, but just to understand... Why are there is many copy pasted static methods instead of an util class (for the new keys method for example)

Simple answer, i was a bit in a hurry (as always) and i had some vague memories that the set/map classes had a similar problem with shared code in the past. But i was not able to find the code so i decide to start with this copy and have a look at this later to get some feedback first. (Greg an i are working (i guess) in different time zones)

File: testsrc/org/mozilla/javascript/tests/DelegatorTest.java [hunk 1]
```diff
@@ -109,6 +109,134 @@ public void setPrototypeOf() {
...
+                + "result += map.has(TestDelegator);\n"
+                + "result += '#';\n"
+
+                + "result += map.get(TestDelegator);\n"
+                + "result += '#';\n"
+
+                + "result += map.delete(TestDelegator);\n"
+                + "result += '#';\n"
+                + "result;";
+        // assertEquals("true#test#true#", eval(script));
```
Comment (SebastianGlass) [inline testsrc/org/mozilla/javascript/tests/DelegatorTest.java:125]: There is an unused assertion. Surely a mistake ;)

**Pull request merged**